pokemon/src/main/java/gamarket/Player.java

```diff
@@ -0,0 +1,242 @@
+    private Grid grid;
+    private PokemonCollection pokeTeam;
+    private PokemonCollection pokeDex;
+
+    /**
+     * Player constructor initializes either a new player or returning player
+     * @param newUser lets the constructor know whether it's a new user or not
+     * @param un gives the constructor the player's username
+     * @param pw gives the constructor the player's password
+     **/
+    public Player(Boolean newUser, String un, String pw) {
+        if (newUser) {
+            Date originalDate = new Date();
+            //format
+            this.joinDate = originalDate.toString();
+            this.name = un;
+            this.password = pw;
+            this.badges = 0;
+            this.money = 0;
+            this.totalPokemon = 0;
+            SimpleDateFormat formatter = new SimpleDateFormat("MM/dd/yyyy");
+            this.joinDate = formatter.format(originalDate);
+            this.grid = new Grid();
+            formatter = new SimpleDateFormat("HH:mm:ss");
+            this.time1 = formatter.format(originalDate);
+        } else {
+            this.name = un;
+            this.password = pw;
+            loadData(this.name);
+
+        }
+    }
+
+    public void setName(String username) {
+        this.name = username;
+    }
+
+    public void setEmail(String e) {
+        this.email = e;
+    }
+
+    public void setPassword(String pw) {
+        this.password = pw;
+    }
+
+    public void setBadges(int amt){
+        this.badges = amt;
+    }
+
+    public void setMoney(long amt){ this.money = amt; }
+
+    public void setTotalPokemon(int amt){ this.totalPokemon = amt; }
+
+    public String getName(){ return name; }
+
+    public String getEmail(){ return email; }
+
+    public String getPassword(){ return password; }
+
+    public int getBadges(){ return badges; }
+
+    public double getMoney(){ return money; }
+
+    public int getTotalPokemon(){ return totalPokemon; }
+
+    public PokemonCollection getPokeTeam(){ return pokeTeam; }
+
+    public PokemonCollection getPokeDex(){ return pokeDex; }
+
+    /**
+     * loadData loads the returning player's saved data
+     * @param fileName takes in the file name where the player's profile is saved
+     *                 Proper player profile format:
+     *                 playerName,email,password
+     *                 amountOfBadges,money,totalPokemon
+     *                 joinDate,totalTime,gridFileName
+     *                 pokeTeamFile,pokedexFile,
+     **/
+    public void loadData(String fileName) {
+        String filePath = "./pokemon/databaseFiles/userProfiles/" + fileName + "_profile.txt";
+        File inFile = new File(filePath);
+
+        Scanner scanner = null;
+        try {
+            scanner = new Scanner(inFile);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+        int line = 1;
+        while (scanner.hasNextLine()) {
```

Comment:
think moveing while block into the try block on line 108 will fix the testing issue

Reply:
it did not fix it